dist_tables/dml.rst

```diff
@@ -38,6 +38,105 @@ For example:
+         count(*) AS view_count
+  FROM page_views
+  GROUP BY view_time::date, site_id, url;
+
+One disadvantage of repeatedly executing the aggregate query is that it always has to recompute on the whole dataset. It's faster to save the information to a roll up :code:`daily_page_views` table and query that. We create the table and use :code:`INSERT INTO ... SELECT` to populate the table directly from the results of a query:
+
+.. code-block:: postgresql
+
+  CREATE TABLE daily_page_views (
+    site_id int,
+    day date,
+    url text,
+    view_count bigint,
+    PRIMARY KEY (site_id, day, url)
+  );
+
+  INSERT INTO daily_page_views (day, site_id, url, view_count)
+    -- this is our original select query
+    SELECT view_time::date AS day, site_id, url,
+           count(*) AS view_count
+    FROM page_views
+    GROUP BY view_time::date, site_id, url;
+
+  -- now the results are available right out of the table
+  SELECT day, site_id, url, view_count
+  FROM daily_page_views;
+
+
+Unfortunately as the numbers of tracked web sites and their pages grow, the aggregate query for populating (or as we'll see later, updating) the rollup table slows down. To compensate we can combine rollup tables and distributed computing to scale the application. Since we GROUP BY each site separately we can parallelize the computation across nodes in a distributed database. Each node in a Citus cluster can hold the data for different web sites and the nodes can each compute rollups locally and write a corresponding part of :code:`daily_page_views`. We distribute both :code:`page_views` and :code:`daily_page_views` so that their rows will stay on the same machine when their :code:`site_id` values match.
+
+.. code-block:: postgresql
+
+  -- First distribute the tables. Notice how we're using the
+  -- same distribution column to keep page views and their daily
+  -- summaries on the same machine
+
+  SELECT create_distributed_table('page_views', 'site_id');
+  SELECT create_distributed_table('daily_page_views', 'site_id');
+
+Keeping the tables' information together on each node, i.e. `co-locating <colocation_groups>`_ them, minimizes network traffic between nodes and allows highly parallel execution. In fact for INSERT INTO SELECT to work in Citus, colocation isn't just a good idea, it's the law. Citus requires the source and destination table to be colocated and throws an error if they are not. Citus implements INSERT INTO SELECT by pushing down the select query to each shard. The distributed query execution happens automatically, just use the ordinary SQL command.
+
+.. code-block:: postgresql
+
+  -- Then run the ordinary INSERT INTO SELECT as before
+
+  INSERT INTO daily_page_views (day, site_id, url, view_count)
+    SELECT view_time::date AS day, site_id, url,
+           count(*) AS view_count
+    FROM page_views
+    GROUP BY view_time::date, site_id, url;
+
+
+In summary, INSERT INTO SELECT on Citus requires that:
```

Comment:
(moving from two paragraphs above to here)

It's worth noting that for INSERT INTO SELECT to work on distributed tables, Citus requires the source and destination table to be colocated. In summary: ...